sklearn/utils/tests/test_typedefs.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+import pytest
+
+from sklearn.utils._typedefs import make_array_from_typed_val
+
+
+@pytest.mark.parametrize(
+    "dtype_t, value, expected_dtype",
+    [
+        ("bool_t", 1, np.uint8),
```

Comment:
``bool_t`` was probably a bad choice since there's no ``np.bool``. We probably should have named it ``uint8_t``. It's only used in hierarchical_fast so I can change it quickly.

@jjerphan, you mentionned that during the review of a previous PR where I changed the uses of ``int8`` in _hierarchical_fast. With this new context, I changed my mind and now think that we should use `ìnt8` consistently. What do you think ?

Reply:
I do not remember exactly, do you mean `uint8` or do you mean `int8`?

I think we tend to use `uint8` in the code-base generally for booleans.

Reply:
The current proposal for boolean on this line looks good to me.

Reply:
+1 for a more precise `uint8_t` type alias instead of `bool_t` which an be misleading.

Reply:
Thinking again, I have a slight preference for `uint8_t` over `bool_t`.